### PR TITLE
PLFM 5709: More flexible kinesis config

### PR DESCRIPTION
**Pull request by marcomarasca**

### CODE REVIEW

File: src/main/resources/templates/repo/kinesis-log-streams.json.vpt [hunk 1]
```diff
@@ -44,6 +44,15 @@
...
+								"Action": [
+									"glue:GetTableVersions"
+								],
+								"Resource":[
+									"*"
+								]
+							},
 							{
 								"Effect":"Allow",
 								"Action":[
```
Comment (marcomarasca) [inline src/main/resources/templates/repo/kinesis-log-streams.json.vpt]: Check with Ziming why we need the kms;Decrypt

File: src/main/resources/templates/repo/kinesis-log-streams.json.vpt [hunk 2]
```diff
@@ -59,27 +68,157 @@
 			]
 		}
 	},
-	#foreach( $kinesisFirehoseStreamName in $kinesisFirehoseStreamNames)
-	"${kinesisFirehoseStreamName}KinesisStream":{
+	#set ($gluePrefix = "${stack}${instance}")
+	#set ($gluePrefix = "${gluePrefix.toLowerCase()}")
+	#set ($glueDatabaseName = "${gluePrefix}firehoselogs")
+	#macro( glueTableName $name )${gluePrefix}${name.toLowerCase()}#end
```
Comment (marcomarasca) [inline src/main/resources/templates/repo/kinesis-log-streams.json.vpt]: Put this processing in the config post processing for the velocity context

File: src/main/resources/templates/repo/kinesis-log-streams.json.vpt [hunk 3]
```diff
@@ -59,27 +68,157 @@
...
+					"Prefix":"${stream.name}/backup/records/${stream.partitionScheme}/",
+					"ErrorOutputPrefix":"${stream.name}/backup/errors/!{firehose:error-output-type}/!{timestamp:yyyy-MM-dd}/",
+					"BufferingHints":{
+						"IntervalInSeconds":${stream.bufferFlushInterval},
+						"SizeInMBs":${stream.bufferFlushSize}
+					}
 				}
+				#elseif( $stream.format == "JSON" )
+				"CompressionFormat":"GZIP"
+				#end
```
Comment (marcomarasca) [inline src/main/resources/templates/repo/kinesis-log-streams.json.vpt:212]: Maybe add some exception function to the context

File: src/test/java/org/sagebionetworks/template/repo/kinesis/KinesisFirehoseConfigValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+		stream.setFormat(KinesisFirehoseRecordFormat.PARQUET);
+		stream.setTableDescriptor(table);
+
+		when(mockConfig.getStreamDescriptors()).thenReturn(Collections.singleton(stream));
+
+		validator.validate();
+	}
+
+	@Test(expected = IllegalStateException.class)
+	public void testWithMissingTableName() {
```
Comment (marcomarasca) [inline src/test/java/org/sagebionetworks/template/repo/kinesis/KinesisFirehoseConfigValidatorTest.java]: Rename method to testWithMissingTableDescriptor

File: src/test/java/org/sagebionetworks/template/repo/kinesis/KinesisFirehoseConfigValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+		KinesisFirehoseStreamDescriptor stream = new KinesisFirehoseStreamDescriptor();
+		stream.setBufferFlushInterval(KinesisFirehoseStreamDescriptor.MAX_BUFFER_INTERVAL + 1);
+
+		when(mockConfig.getStreamDescriptors()).thenReturn(Collections.singleton(stream));
+
+		validator.validate();
+
+	}
+
```
Comment (marcomarasca) [inline src/test/java/org/sagebionetworks/template/repo/kinesis/KinesisFirehoseConfigValidatorTest.java:116]: Missing minBufferSizeTest

**Review by john-hill — APPROVED** [review]

**Pull request merged**